plugins/influxdb/influxdb.go

```diff
@@ -0,0 +1,182 @@
+package influxdb
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-go/statsd"
+	"github.com/Sirupsen/logrus"
+	"github.com/stripe/veneur/plugins"
+	"github.com/stripe/veneur/samplers"
+)
+
+var _ plugins.Plugin = &InfluxDBPlugin{}
+
+// InfluxDBPlugin is a plugin for emitting metrics to InfluxDB.
+type InfluxDBPlugin struct {
+	Logger     *logrus.Logger
+	InfluxURL  string
+	HTTPClient *http.Client
+	Statsd     *statsd.Client
+}
+
+// NewInfluxDBPlugin creates a new Influx Plugin.
+func NewInfluxDBPlugin(logger *logrus.Logger, addr string, consistency string, db string, client *http.Client, stats *statsd.Client) *InfluxDBPlugin {
+	plugin := &InfluxDBPlugin{
+		Logger:     logger,
+		HTTPClient: client,
+		Statsd:     stats,
+	}
+
+	inurl, err := url.Parse(addr)
+	if err != nil {
+		logger.Fatalf("Error parsing URL for InfluxDB: %q", err)
+	}
+
+	// Construct a path we will be using later.
+	inurl.Path = "/write"
+	q := inurl.Query()
+	q.Set("db", db)
+	q.Set("precision", "s")
+	inurl.RawQuery = q.Encode()
+	plugin.InfluxURL = inurl.String()
+
+	return plugin
+}
+
+// Flush sends a slice of metrics to InfluxDB
+func (p *InfluxDBPlugin) Flush(metrics []samplers.DDMetric, hostname string) error {
+	p.Statsd.Gauge("flush.post_metrics_total", float64(len(metrics)), nil, 1.0)
+	// Check to see if we have anything to do
+	if len(metrics) == 0 {
+		p.Logger.Info("Nothing to flush, skipping.")
+		return nil
+	}
+
+	buff := bytes.Buffer{}
+	colons := regexp.MustCompile(":")
+	for _, metric := range metrics {
+		tags := strings.Join(metric.Tags, ",")
+		cleanTags := colons.ReplaceAllLiteralString(tags, "=")
+		buff.Write(
+			[]byte(fmt.Sprintf("%s,%s value=%f %d\n", metric.Name, cleanTags, metric.Value[0][1], int64(metric.Value[0][0]))),
+		)
+	}
+
+	p.postHelper(p.InfluxURL, &buff, "fart", false)
+
+	return nil
+}
+
+// Name returns the name of the plugin.
+func (p *InfluxDBPlugin) Name() string {
+	return "influxdb"
+}
+
+// shared code for POSTing to an endpoint, that consumes JSON, that is zlib-
+// compressed, that returns 202 on success, that has a small response
+// action is a string used for statsd metric names and log messages emitted from
+// this function - probably a static string for each callsite
+// you can disable compression with compress=false for endpoints that don't
+// support it
+func (p *InfluxDBPlugin) postHelper(endpoint string, bodyBuffer *bytes.Buffer, action string, compress bool) error {
+	// attach this field to all the logs we generate
```

Comment:
`action` could probably be a const defined within this function, since it's only used once.